docs/concepts.md

```diff
@@ -0,0 +1,311 @@
+# Key Concepts
+
+In order to make your interaction with `ralph-cli` easier, we need to establish
+some common vocabulary - hence the following section, where key concepts behind
+`ralph-cli` are explained.
+
+## Config
+
+`ralph-cli` has its own configuration file `~/.ralph-cli/config.toml`, in
+[TOML][] format. This file is created automatically at the first run of
+`ralph-cli`, and it contains some default settings, as presented below:
+
+```no-highlight
+Debug = false
+LogOutput = ""
+ClientTimeout = 10
+RalphAPIURL = "change_me"
+RalphAPIKey = "change_me"
+ManagementUserName = "change_me"
+ManagementUserPassword = "change_me"
+```
+
+The first three of them (`Debug`, `LogOutput` and `ClientTimeout`) are not used
+at this moment, so for now, you can safely ignore their existence (although we
+are going to turn them on quite soon). Quite important are the ones with dummy
+values denoted by `"change_me"` string - these are required for `ralph-cli` to
+operate, and their meaning is as follows:
+
+* `RalphAPIURL` - this should be a string with an URL to your Ralph instance,
+  with `/api` path appended, e.g. `"https://my-ralph-instance.local/api"`
+* `RalphAPIKey` - your personal API key that `ralph-cli` should use for
+  accessing Ralph; you can find this key by visiting Ralph's GUI (go to the
+  top-right corner, click on your user name and then select "My profile" - in
+  "Personal info" section you should see a field named "API Token" - that's it).
+* `ManagementUserName` - user name that is used to access iDRAC/iLO, which
+  `ralph-cli` exposes to scan scripts as `MANAGEMENT_USER_NAME` environment
+  variable (see [Scripts Contract][self-contract])
+* `ManagementUserPassword` - password for the above, exposed as
+  `MANAGEMENT_USER_PASSWORD` environment variable
+
+Please note that due to the presence of credentials in `config.toml`, this file
+should remain readable only to its owner (it is `0600` by default, so you don't
+have to do anything) - otherwise `ralph-cli` will refuse to cooperate.
+
+Also, keep in mind that this is just an initial structure of `ralph-cli` config
+file, and it is subject to change until version `1.0.0` is reached.
+
+## Scan
```

Comment:
What fragment do you have in mind..? If "Scan", then no - it's one of the concepts, therefore it should explained here. When we will add other commands, e.g. `deploy`, then yes, that will make sense, but at this point I can't see no real gain here.